include/seastar/core/sstring.hh

```diff
@@ -597,7 +597,7 @@ public:
     }
 
     constexpr bool starts_with(std::basic_string_view<char_type, traits_type> sv) const noexcept {
-        return size() > sv.size() && compare(0, sv.size(), sv) == 0;
+        return size() >= sv.size() && compare(0, sv.size(), sv) == 0;
```

Comment:
As in the cover letter, please also rename this patch, and add some text explaining what you did (i.e., fix the behavior of these functions when given two identical strings - x.starts_with(x) should be true, not false.
"same as std::string" is much too general.

Reply:
ok, have re-pushed and changed the PR's title.

Reply:
Thanks.